### PR TITLE
PWN-8470 - fixed problems related to disconnected inet

### PR DESCRIPTION
**Pull request by eduardmaximovich**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8470

## Description of Work

- Reworked `ConnectionManager` - we can no longer rely on system statuses provided by ConnectivityManager, as it do not accurately display the factual state
- Fixed reconnecting to websockets

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/updates/SocketUpdatesManager.kt [hunk 1]
```diff
@@ -37,7 +38,7 @@ private const val TAG = "Sockets:SocketUpdatesManager"
 class SocketUpdatesManager(
     appScope: AppScope,
     private val environmentManager: NetworkEnvironmentManager,
-    private val connectionStateProvider: NetworkConnectionStateProvider,
+    private val connectionStateProvider: ConnectionManager,
```
Comment (eduardmaximovich) [inline app/src/main/java/org/p2p/wallet/updates/SocketUpdatesManager.kt:41]: ConnectionManager measures the internet state more accurately so i've changed it

File: app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt [hunk 2]
```diff
@@ -68,6 +69,8 @@ class UserTokensPolling(
                 startPolling()
             } catch (e: CancellationException) {
                 Timber.i("Cancelled tokens remote update")
+            } catch (e: UnknownHostException) {
+                Timber.i(e, "Failed polling tokens: no internet connection")
```
[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt:73]: we are already filtering this in SentryFacade, you could check that

File: app/src/main/java/org/p2p/wallet/updates/SocketUpdatesManager.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/updates/SocketUpdatesManager.kt:41]: should we delete NetworkConnectionStateProvider?

File: app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt:73]: You're right. But what about firebase? It seems FirebaseCrashlyticsFacade doesn't filter such things, or is it okay?

[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt:73]: Yep, Firebase doesn't have any quota on events, so it's okay

File: app/src/main/java/org/p2p/wallet/updates/SocketUpdatesManager.kt (hunk 1 above)
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/updates/SocketUpdatesManager.kt:41]: I assume yes, connectivity manager doesn't check the internet in many cases, sometimes even if we turned off all transports it still saying that everything okay. But let's keep it for a while until the testers have tested current bug.

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**